network/netplan/netplan_test.go

```diff
@@ -0,0 +1,605 @@
+    metric: 3
+`[1:]
+	var np netplan.Netplan
+	err := netplan.Unmarshal([]byte(input), &np)
+	out, err := netplan.Marshal(np)
+	c.Check(err, jc.ErrorIsNil)
+	c.Check(string(out), gc.Equals, input)
+}
+
+func (s *NetplanSuite) TestSimpleBridger(c *gc.C) {
+	input := `
+network:
+  version: 2
+  renderer: NetworkManager
+  ethernets:
+    id0:
+      match:
+        macaddress: "00:11:22:33:44:55"
+      addresses:
+      - 1.2.3.4/24
+      - 2000::1/64
+      gateway4: 1.2.3.5
+      gateway6: 2000::2
+      nameservers:
+        search: [foo.local, bar.local]
+        addresses: [8.8.8.8]
+`[1:]
+	expected := `
+network:
+  version: 2
+  renderer: NetworkManager
+  ethernets:
+    id0:
+      match:
+        macaddress: "00:11:22:33:44:55"
+  bridges:
+    juju-bridge:
+      interfaces: [id0]
+      addresses:
+      - 1.2.3.4/24
+      - 2000::1/64
+      gateway4: 1.2.3.5
+      gateway6: 2000::2
+      nameservers:
+        search: [foo.local, bar.local]
+        addresses: [8.8.8.8]
+`[1:]
+	var np netplan.Netplan
+
+	err := netplan.Unmarshal([]byte(input), &np)
+	c.Check(err, jc.ErrorIsNil)
+
+	err = np.BridgeEthernetById("id0", "juju-bridge")
+	c.Check(err, jc.ErrorIsNil)
+
+	out, err := netplan.Marshal(np)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(string(out), gc.Equals, expected)
+}
+
+func (s *NetplanSuite) TestBridgerIdempotent(c *gc.C) {
+	input := `
+network:
+  version: 2
+  renderer: NetworkManager
+  ethernets:
+    id0:
+      match:
+        macaddress: "00:11:22:33:44:55"
+  bridges:
+    juju-bridge:
+      interfaces: [id0]
+      addresses:
+      - 1.2.3.4/24
+      - 2000::1/64
+      gateway4: 1.2.3.5
+      gateway6: 2000::2
+      nameservers:
+        search: [foo.local, bar.local]
+        addresses: [8.8.8.8]
+`[1:]
+	var np netplan.Netplan
+	err := netplan.Unmarshal([]byte(input), &np)
+	c.Check(err, jc.ErrorIsNil)
+	err = np.BridgeEthernetById("id0", "juju-bridge")
+	c.Check(err, jc.ErrorIsNil)
+	out, err := netplan.Marshal(np)
+	c.Check(string(out), gc.Equals, input)
+
+}
+
+func (s *NetplanSuite) TestBridgerBridgeExists(c *gc.C) {
+	input := `
+network:
+  version: 2
+  renderer: NetworkManager
+  ethernets:
+    id0:
+      match:
+        macaddress: "00:11:22:33:44:55"
+      addresses:
+      - 1.2.3.4/24
+      - 2000::1/64
+      gateway4: 1.2.3.5
+      gateway6: 2000::2
+      nameservers:
+        search: [foo.local, bar.local]
+        addresses: [8.8.8.8]
+    id1:
+      match:
+        driver: ixgbe
+  bridges:
+    juju-bridge:
+      interfaces: [id1]
+      addresses:
+      - 1.2.3.4/24
+      - 2000::1/64
+      gateway4: 1.2.3.5
+      gateway6: 2000::2
+      nameservers:
+        search: [foo.local, bar.local]
+        addresses: [8.8.8.8]
+`[1:]
+	var np netplan.Netplan
+	err := netplan.Unmarshal([]byte(input), &np)
+	c.Check(err, jc.ErrorIsNil)
+	err = np.BridgeEthernetById("id0", "juju-bridge")
+	c.Check(err, gc.ErrorMatches, "Bridge named \"juju-bridge\" already exists")
```

Comment:
I think we need a little more context in the error, so that we can understand the context of the failure (such as something about 'id0')